### ISSUE TITLE
UnhandledPromiseRejectionWarning: Error: EEXIST: file already exists, mkdir  

### ISSUE THREAD

**Issue by swyxio** [DanWebb/jdown#17]
I don't know what causes this but for some reason the `rmrf` inside `transform-assets` fails, and so the `mkdir` silently fails, and then there is an `UnhandledPromiseRejectionWarning`:

```
(node:34749) UnhandledPromiseRejectionWarning: Error: EEXIST: file already exists, mkdir '/Users/swyx/Work/swyxdotio/static/jdown-assets/content'
(node:34749) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 1)
(node:34749) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
```

**Comment by DanWebb** [comment]
Hey @sw-yx sorry to hear you're having trouble with this.

Based on the [rimraf readme](https://github.com/isaacs/rimraf#api) there's a few filesystem errors it looks like it might run into.

Could you try running rimraf from the command line like `npx rimraf jdown-assets/content` (presuming you `cd` into the `/Users/swyx/Work/swyxdotio/static/` dir).

Hopefully, that will show the error you're running into then we'll have a better idea of what's going on 🙏

If it doesn't error then could you provide your node version, operating system and the exact string you use for the [assets output option](https://github.com/DanWebb/jdown#assets) for me? That should give me a better starting point in trying to replicate it.

**Comment by swyxio** [comment]
no error at all (just like when i use rimraf inside jdown, also no error)

this is the exact config i used:

```js
const jdown = require('jdown') // parses a folder of markdown files into objects. very handy!
const content = await jdown('content', {
  // so that markdown images show nicely
  assets: {
    output: './static/jdown-assets',
    path: '/jdown-assets',
  },
})
```

but ultimately i think the easier solve here is to just wrap that mkdir call in a try/catch since you can't be sure that the rimraf succeeds, and you're DEFINITELY calling two mkdir's in a row in some cases there

**Comment by DanWebb** [comment]
@sw-yx I just published v1.0.3 which adjusts the directory creation to automatically recursively create the folders if they don't exist using the handy [make-dir](https://www.npmjs.com/package/make-dir). 

That should fix the mkdir error you're seeing at least.

Still, I'm afraid if rimraf is failing for some reason on your system then we may see issues with duplicate assets being created. If that does end up being the case I'll need to request your node version and operating system again so I can have a chance at finding what might be causing it.

**Comment by swyxio** [comment]
ah sorry i forgot to put it there the first time. node 10 and mac osx. am on mobile now so i cant double check but happy to close this until someone else or me runs into this
